src/compiler/transformers/module/module.ts

```diff
@@ -214,20 +214,28 @@ namespace ts {
         function addExportEqualsIfNeeded(statements: Statement[], emitAsReturn: boolean) {
             if (exportEquals && resolver.isValueAliasDeclaration(exportEquals)) {
                 if (emitAsReturn) {
-                    statements.push(createReturn(exportEquals.expression));
+                    const statement = createReturn(
+                        exportEquals.expression,
+                        /*location*/ exportEquals
+                    );
+
+                    setNodeEmitFlags(statement, NodeEmitFlags.NoTokenSourceMaps | NodeEmitFlags.NoComments);
+                    statements.push(statement);
                 }
                 else {
-                    statements.push(
-                        createStatement(
-                            createAssignment(
-                                createPropertyAccess(
-                                    createIdentifier("module"),
-                                    "exports"
-                                ),
-                                exportEquals.expression
-                            )
-                        )
+                    const statement = createStatement(
+                        createAssignment(
+                            createPropertyAccess(
+                                createIdentifier("module"),
+                                "exports"
+                            ),
+                            exportEquals.expression
+                        ),
+                        /*location*/ exportEquals
                     );
+
+                    setNodeEmitFlags(statement, NodeEmitFlags.NoComments);

```

Comment:
Why not make `NodeEmitFlags` a parameter to `create*`? It seems like it would be a lot harder to forget. If the majority of uses don't need to call `setNodeEmitFlags`, then maybe not. I just saw lots of calls to it in this PR.